include/ginkgo/core/stop/residual_norm.hpp

```diff
@@ -203,12 +210,18 @@ class RelativeResidualNorm : public ResidualNorm<ValueType> {
 
         auto exec = factory->get_executor();
 
-        auto dense_rhs = as<Vector>(args.b);
         this->starting_tau_ =
             NormVector::create(exec, dim<2>{1, args.b->get_size()[1]});
         this->u_dense_tau_ =
             NormVector::create_with_config_of(this->starting_tau_.get());
-        dense_rhs->compute_norm2(this->starting_tau_.get());
+        if (dynamic_cast<const matrix::Dense<to_complex<ValueType>> *>(
+                args.b.get())) {
+            auto dense_rhs = as<matrix::Dense<to_complex<ValueType>>>(args.b);
+            dense_rhs->compute_norm2(this->starting_tau_.get());
+        } else {
+            auto dense_rhs = as<Vector>(args.b);
+            dense_rhs->compute_norm2(this->starting_tau_.get());
+        }
```

Comment:
I am confused about these codes.
Could you explain the difference between if-else?

Reply:
When you use IDR with a complex subspace for a real ValueType, then `as<Vector>` expects a real vector for the `args.initial_residual`, but it is a complex vector. Also the system matrix has a different ValueType than the residual in this case, so I need to distinguish here.

Reply:
I see. Thanks